File: Compiler/bin/Debug/CanvasEngine/Renderer2.js

/*
 * Build with SharpToJs Compiler
 * Author: Vito Domenico Tagliente
 */


CanvasEngine.Graphics.Renderer2 = function(device, GraphicDevice){

	this.Device;
	this.style = "fill";
	this.color = "black";

	Object.defineProperty( this, 'Context', {
		get: function()
		{
			return this.Device.Context;
		}
	} );

	this.Begin = function()
	{
		this.Context.beginPath(  );
	}

	this.Save = function()
	{
		this.Context.save(  );
	}

	this.Restore = function()
	{
		this.Context.restore(  );
	}

	this.End = function()
	{
		if ( this.style == "fill" )
		{
			this.Context.fill(  );
			return;
		}

		this.Context.stroke(  );
	}

	this.Rect = function(point2, Point2, width, height)
	{
		if ( this.style == "fill" )
		{
			this.Context.fillRect( point2.x, point2.y, width, height );
			return;
		}

		this.Context.strokeRect( point2.x, point2.y, width, height );
	}

	// Constructor

	this.Device = device;


}

File: Compiler/bin/Debug/CanvasEngine/Renderer2.cs
using System;
using SharpToJs;

namespace CanvasEngine.Graphics
{
	public class Renderer2
	{
		public GraphicDevice Device;
		public string style = "fill";
		public string color = "black";
		
		public JsNative Context {
			get {
				return Device.context;
			}
		}
		
		public Renderer2(GraphicDevice device)
		{
			Device = device;
		}
		
		public void Begin()
		{
			Context.beginPath()
		}
		
		public void Save()
		{
			Context.save();
		}
		
		public void Restore()
		{
			Context.restore();
		}
		
		public void End()
		{
			if( style == "fill" ){
				Context.fill();
				return;
			}
			Context.stroke();
		}
		
		public void Rect( Point2 point2, int width, int height )
		{
			if(style == "fill"){
				Context.fillRect( point2.x, point2.y, width, height );
				return;
			}
			Context.strokeRect( point2.x, point2.y, width, height );
		}
		
	}
}